### PR TITLE
Search changes

### PR DESCRIPTION
**Pull request by TiruDeepthi**
PR for search API and bug fixes

### CODE REVIEW

File: src/components/NormalizationDashboard.jsx [hunk 1]
```diff
@@ -30,13 +31,35 @@ function NormalizationDashboard() {
...
+      method: "POST",
+      headers: {
+        "Content-type": "application/json; charset=UTF-8"
+      }
+    })
+      .then((response) => response.json())
+      .then((result) => {
+        setCustomers(result?.data);
+        setLoader(false);
+      });
```
Comment (RajKumar-BackCountry) [inline src/components/NormalizationDashboard.jsx]: how are u handling errors ?
Also why `then`.
Use await.

File: src/components/Table.jsx [hunk 2]
```diff
@@ -27,6 +27,9 @@ import useSessionStorage from "../hooks/useSessionStorage";
 import { status } from "../constants/index";
 import AssignToEditor from "../logos/AssignToEditor.svg";
 import AssignToWriter from "../logos/AssignToWriter.svg";
+import ArrowSortDownLine from "../logos/ArrowSortDownLine.svg";
+import ArrowSortUpLine from "../logos/ArrowSortUpLine.svg";
+import { Tooltip } from 'primereact/tooltip';
```
Comment (RajKumar-BackCountry) [inline src/components/Table.jsx]: Import from libraries should be at top ahead of file imports.

Follow same standards everywhere.

File: src/components/Table.jsx [hunk 3]
```diff
@@ -72,7 +77,30 @@ export default function Table() {
...
+      case "Assigned":
+        status.push("ASSIGNED_TO_WRITER", "ASSIGNED_TO_EDITOR");
+        break;
+      case "InProgress":
+        status.push("WRITING_IN_PROGRESS", "EDITING_IN_PROGRESS");
+        break;
+      default:
+        status;
+    }
+    const body = {
```
Comment (RajKumar-BackCountry) [inline src/components/Table.jsx]:
```
const body = {
    filters: { status }
}
```

File: src/components/Table.jsx [hunk 4]
```diff
@@ -72,7 +77,30 @@ export default function Table() {
...
+      filters: {
+        status:status
+      }
+    }
+
+    fetch(`${workFlowsUrl}/search?limit=999&page=${currentPage}&unique=brand`, {
```
Comment (RajKumar-BackCountry) [inline src/components/Table.jsx]: dont hard code 999.
Keep it in  constants file.

File: src/components/Table.jsx [hunk 5]
```diff
@@ -103,13 +131,14 @@ export default function Table() {
 
     fetch(`${workFlowsUrl}/search?limit=10&page=${currentPage}&unique=status`, {
       method: "POST",
+      body: JSON.stringify(body),
       headers: {
         "Content-type": "application/json; charset=UTF-8"
       }
     })
       .then((response) => response.json())
       .then((result) => setStatus(result?.data?.uniqueValues));
```
Comment (RajKumar-BackCountry) [inline src/components/Table.jsx]: same as above. 
use await and handle errors.

File: src/components/Table.jsx [hunk 6]
```diff
@@ -461,39 +501,47 @@ export default function Table() {
       </span>
     );
   };
+
   const handleAssigneeSort = () => {
```
Comment (RajKumar-BackCountry) [inline src/components/Table.jsx]: Too much duplicated code.
Merge all of these methods into a single function call.

File: src/components/Table.jsx [hunk 7]
```diff
@@ -522,28 +570,46 @@ export default function Table() {
...
+            <Tooltip target=".quick-fix"/>
+            <img alt={`${Edit} svg`} src={Edit} data-pr-tooltip="Quick Fix" data-pr-position="top" className="quick-fix"  style={{ fontSize: '200px', cursor: 'pointer' }}></img>
+            </button> 
+            :
+            <button
+            className="bg-white flex rounded-full justify-center items-center border border-grey-30  h-[30px] w-[30px]"
+            onClick={handleEditIcon}
+          >
+            <Tooltip target=".assign"/>
+            <img alt={`${ currentTab == "Assigned" || currentTab == "InProgress"
```
Comment (RajKumar-BackCountry) [inline src/components/Table.jsx]: Your prettier alignment is wrong.
Fix the vscode issue and make sure prettier works on save.

File: src/components/Table.jsx [hunk 8]
```diff
@@ -522,28 +570,46 @@ export default function Table() {
...
+                      ? ReAssign
+                      : AssigneEdit} svg`} src={ currentTab == "Assigned" || currentTab == "InProgress"
+                      ? ReAssign
+                      : AssigneEdit} 
+                      data-pr-tooltip="Assign" data-pr-position="top"  className="assign"  style={{ fontSize: '200px', cursor: 'pointer' }}
```
Comment (RajKumar-BackCountry) [inline src/components/Table.jsx]: Replace these styles with tailwind styles.

File: src/components/Table.jsx (hunk 3 above)
Comment (TiruDeepthi) [reply src/components/Table.jsx]: addressed

File: src/components/Table.jsx (hunk 2 above)
Comment (TiruDeepthi) [reply src/components/Table.jsx]: addressed

File: src/components/Table.jsx (hunk 4 above)
Comment (TiruDeepthi) [reply src/components/Table.jsx]: addressed

File: src/components/Table.jsx (hunk 5 above)
Comment (TiruDeepthi) [reply src/components/Table.jsx]: addressed

File: src/components/Table.jsx (hunk 6 above)
Comment (TiruDeepthi) [reply src/components/Table.jsx]: addressed

File: src/components/Table.jsx (hunk 8 above)
Comment (TiruDeepthi) [reply src/components/Table.jsx]: addressed

File: src/components/Table.jsx [hunk 9]
```diff
@@ -143,31 +115,119 @@ export default function Table() {
...
-        setLoader(false);
-      });
+  );
+  const data = await response.json();
+  if(data?.success){
+    setCustomers(data?.data);
+    setLoader(false)
+  }
+  if (!data?.success){
+    console.log("data>>",data?.error)
```
Comment (RajKumar-BackCountry) [inline src/components/Table.jsx]: remove console.log

File: src/components/Table.jsx [hunk 10]
```diff
@@ -143,31 +115,119 @@ export default function Table() {
...
+	}
+}
+
+async function getBrands(){
+const response = await fetch(`${workFlowsUrl}/search?limit=${limit}&page=${currentPage}&unique=brand`,
```
Comment (RajKumar-BackCountry) [inline src/components/Table.jsx:184]: why is this approach not implemented in other places.
I can still see some fetch calls with then

File: src/components/Table.jsx [hunk 11]
```diff
@@ -143,31 +115,119 @@ export default function Table() {
...
+          "Content-type": "application/json; charset=UTF-8"
+        }
+      }
+  );
+  const data = await response.json();
+  if(data?.success){
+    setAssignee(data?.data?.uniqueValues.filter(Boolean))
+  }
+  if (!data?.success){
+    console.log("data>>",data?.error)
```
Comment (RajKumar-BackCountry) [inline src/components/Table.jsx]: Invoke the Toast component and pass this text.
"Something went wrong. Please try again"

File: src/components/Table.jsx [hunk 12]
```diff
@@ -461,40 +532,42 @@ export default function Table() {
...
-  const handleStyleSort = () => {
-    setCurrentSort("styleSort");
-    setStyleSort(styleSort == "desc" ? "asc" : "desc");
-  };
+  const handleSort=(currentSort)=>{
+    setCurrentSort(currentSort);
+    switch (currentSort) {
+      case "Style":
+        setStyleSort(styleSort == "desc" ? "asc" : "desc");
+        setDefaultSort(styleSort == "desc" ? ArrowSortDownLine : ArrowSortUpLine)
```
Comment (RajKumar-BackCountry) [inline src/components/Table.jsx:541]: this line is duplicated in all the cases.
move it out of the cases.

File: src/components/Table.jsx (hunk 11 above)
Comment (TiruDeepthi) [reply src/components/Table.jsx]: addressd

File: src/components/Table.jsx (hunk 10 above)
Comment (TiruDeepthi) [reply src/components/Table.jsx:184]: addressed

File: src/components/Table.jsx (hunk 9 above)
Comment (TiruDeepthi) [reply src/components/Table.jsx]: addressed

**Pull request merged**